src/riak_ensemble_peer.erl

```diff
@@ -1802,6 +1844,18 @@ setup({init, Args}, State0=#state{id=Id, ensemble=Ensemble, ets=ETS, mod=Mod}) -
     probe(init, State3).
 
 -spec handle_event(_, atom(), state()) -> {next_state, atom(), state()}.
+handle_event({watch_leader_status, Pid}, StateName, State) when node(Pid) =/= node() ->
+    lager:warning("Remote pid ~p not allowed to watch_leader_status on ensemble peer ~p",
+                  [Pid, State#state.id]),
+    {next_state, StateName, State};
+handle_event({watch_leader_status, Pid}, StateName, State = #state{watchers = Watchers}) ->
+    _ = notify_leader_status(Pid, StateName, State),
+    %% Might as well take this opportunity to prune any dead pids that are in the list
+    NewWatcherList = [P || P <- [Pid | Watchers], is_process_alive(P)],
```

Comment:
Would be cleaner and less likely to send extra messages to dead pids if we just monitored the Pid here and handled a DOWN message, removing the process's Pid from watchers, than to occasionally clean up if someone adds a watcher. Otherwise, I'd also do the same is_process_alive check on the NewWatcherList in stop_watching at least... that said, monitors are there for a reason... seems like we should use them.